### ISSUE TITLE
Add Integration tests

### ISSUE THREAD

**Issue by farbodahm** [DataChefHQ/damavand#36]
Based on experience, best way to test IaC would be to test the final deployment to see if the desired behaviour is reached or not.

Possible solution could be to https://github.com/localstack/localstack to setup a local AWS and deploy resources; Then run assertions on that stack.

**Comment by kkiani** [comment]
@farbodahm That might be a good idea for AWS, but what about Azure? the pulumi tests just support all providers out of the box.

**Comment by farbodahm** [comment]
@kkiani Yeah it was mainly for AWS. Although Azure also provides local stack for lots of the services, the ones that I've worked with:

- https://learn.microsoft.com/en-us/azure/event-hubs/test-locally-with-event-hub-emulator?tabs=automated-script
- https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite?tabs=visual-studio%2Cblob-storage

**Comment by kkiani** [comment]
@farbodahm sounds promising! Let's try it then.
